provider/azure/environ.go

```diff
@@ -627,12 +638,32 @@ func (env *azureEnviron) createVirtualMachine(
 			`[resourceId('Microsoft.Compute/availabilitySets','%s')]`,
 			availabilitySetName,
 		)
+		var availabilitySetProperties interface{}
+		if maybeStorageAccount == nil {
+			// This model uses managed disks; we must create
+			// the availability set as "aligned" to support
+			// them.
+			availabilitySetProperties = &compute.AvailabilitySetProperties{
+				// Managed means the availability set is
+				// "aligned", allowing managed disks to be
+				// used.
+				Managed: to.BoolPtr(true),
+
+				// Azure complains when the fault domain count
```

Comment:
Is there a bug raised for this we can reference here?

Reply:
not yet, not sure where to raise it. I'll try and figure that out...

Reply:
I've left a comment on https://docs.microsoft.com/en-us/rest/api/compute/availabilitysets/availabilitysets-create